lib/wootric/test.js

```diff
@@ -0,0 +1,155 @@
+      .assumesPageview()
+        .option('account_token', '')
+        .global('wootricSettings')
+        .global('wootric_survey_immediately')
+        .global('wootric')
+    )
+  });
+
+  describe('before loading', function(){
+    beforeEach(function(){
+      analytics.stub(wootric, 'load');
+    });
+
+    it('should not have a wootric object', function(){
+      analytics.assert(!window.wootric);
+    })
+
+    describe('#initialize', function(){
+      beforeEach(function(){
+        analytics.initialize();
+        analytics.page();
+      });
+
+      it ('should setup the wootricSettings object', function(){
+        analytics.assert(window.wootricSettings);
+        analytics.assert(window.wootricSettings instanceof Object)
+      })
+
+      it('should call #load', function(){
+        analytics.called(wootric.load);
+      });
+    });
+  });
+
+  describe('loading', function(){
+    it('should load', function(done){
+      analytics.load(wootric, done);
+    });
+  });
+
+  describe('after loading', function(){
+    beforeEach(function(done){
+      analytics.once('ready', done);
+      analytics.initialize();
+      analytics.page();
+    });
+
+    it('should have created the global wootric object', function(){
+      analytics.assert(window.wootric instanceof Function);
+    })
+
+    it('should have settings with account token', function(){
+      analytics.assert(window.wootricSettings.account_token == 'NPS-01fe3cbc')
+    })
+
+    describe('#identify', function(){
+
+      it('should set email on identify', function(){
+        analytics.identify({
+          email: 'shawn@shawnmorgan.com'
+        });
+        analytics.assert(window.wootricSettings.email, 'shawn@shawnmorgan.com');
+      });
+
+      it('should set created_at on identify using ISO MM/DD/YYYY format', function(){
+        analytics.identify({
+          createdAt: '01/01/2015'
+        });
+        analytics.assert(window.wootricSettings.created_at, '1420099200000');
+      });
+
+      it('should set created_at on identify using ISO YYYY-MM-DD format', function(){
+        analytics.identify({
+          createdAt: '2015-01-01'
+        });
+        analytics.assert(window.wootricSettings.created_at, '1420099200000');
+      });
+
+      it('should set created_at on identify using ISO YYYYMMDD format', function(){
+        analytics.identify({
+          createdAt: '20150101'
+        });
+        analytics.assert(window.wootricSettings.created_at, '1420099200000');
+      });
+
+      it('should set created_at on traits using Unix Timestamp format', function(){
+        analytics.identify({
+          createdAt: '1420099200000'
+        });
+        analytics.assert(window.wootricSettings.created_at, '1420099200000');
+      });
+
+      it('should set properties based on other traits', function(){
+        analytics.identify({
+            email: 'shawn@shawnmorgan.com',
+            createdAt: '01/01/2015',
+            properties: {
```

Comment:
Users won't pass these under a `properties` subobject, so you'll need to handle them at the top level.